db/collection_test.go

```diff
@@ -34,35 +35,48 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
-func TestNewCollectionFromInstance(t *testing.T) {
+func TestNewCollection(t *testing.T) {
 	t.Parallel()
 	t.Run("Single", func(t *testing.T) {
 		t.Parallel()
 		db, clean := createTestDB(t)
 		defer clean()
-		_, err := db.NewCollectionFromInstance("Dog", &Dog{})
+		_, err := db.NewCollection(CollectionConfig{
+			Name:   "Dog",
+			Schema: util.SchemaFromInstance(&Dog{}),
+		})
 		checkErr(t, err)
 	})
 	t.Run("Multiple", func(t *testing.T) {
 		t.Parallel()
 		db, clean := createTestDB(t)
 		defer clean()
-		_, err := db.NewCollectionFromInstance("Dog", &Dog{})
+		_, err := db.NewCollection(CollectionConfig{
+			Name:   "Dog",
+			Schema: util.SchemaFromInstance(&Dog{}),
+		})
 		checkErr(t, err)
-		_, err = db.NewCollectionFromInstance("Person", &Person{})
+		_, err = db.NewCollection(CollectionConfig{
+			Name:   "Person",
+			Schema: util.SchemaFromInstance(&Person{}),
+		})
 		checkErr(t, err)
 	})
-	t.Run("Fail/WithoutInstanceID", func(t *testing.T) {
-		t.Parallel()
-		type FailingType struct {
-			IDontHaveAnIDField int
-		}
-		db, clean := createTestDB(t)
-		defer clean()
-		if _, err := db.NewCollectionFromInstance("FailingType", &FailingType{}); err != ErrInvalidCollectionType {
-			t.Fatal("the collection should be invalid")
-		}
-	})
+	// t.Run("Fail/WithoutInstanceID", func(t *testing.T) {
+	// 	t.Parallel()
+	// 	type FailingType struct {
+	// 		IDontHaveAnIDField int
+	// 	}
+	// 	db, clean := createTestDB(t)
+	// 	defer clean()
+	// 	cc := CollectionConfig{
+	// 		Name:   "FailingType",
+	// 		Schema: util.SchemaFromInstance(&FailingType{}),
+	// 	}
+	// 	if _, err := db.NewCollection(cc); err != ErrInvalidCollectionType {
+	// 		t.Fatal("the collection should be invalid")
+	// 	}
+	// })
```

Comment:
This is disabled until we start verifying that the schema specifies ID, tracking here https://github.com/textileio/go-threads/issues/281